system/lib/wasmfs/wasmfs.cpp

```diff
@@ -107,11 +107,11 @@ __wasi_errno_t __wasi_fd_close(__wasi_fd_t fd) {
 
 long __syscall_fstat64(long fd, long buf) {
   // Release FileTable lock after accessing desired open file.
-  std::shared_ptr<File> file = FileTable::get()[fd]->get().getFile();
+  std::shared_ptr<File> file = FileTable::get()[fd].locked().getFile();
 
   struct stat* buffer = (struct stat*)buf;
 
-  auto fileInfo = file->get();
+  auto fileInfo = file->locked();
```

Comment:
How about this?
```suggestion
  auto lockedFile = file->locked();
```
("info" suggests metadata to me, which feels unsuitable perhaps?)

More generally, though, do we need to keep around both a locked and unlocked file here?

Reply:
> For the first point, can we not do lockedFile->is<Directory>()? 

The `is<Directory>()` works on any `File` derived class. But `lockedFile` is a `File::Handle`. I considered putting a utility on `File::Handle` which would basically do the same thing, but I thought it was redundant. Let me know if it would make sense to do so.

Reply:
> if you have a big enough method and you forget how fileInfo was defined higher up, you may be surprised that fileInfo keeps a lock on a file and all its data, and is not just a copy (or reference) to metadata.

This is a good point that I hadn't considered. Makes sense.